### PR TITLE
Add: Unassigned Pull request app

### PR DESCRIPTION
**Pull request by RaiVaibhav**
Add unassigned pull request app, currently it work
only on coala/coala repo.

### CODE REVIEW

**Comment by RaiVaibhav** [comment]
currently in wip

**Comment by RaiVaibhav** [comment]
currenlty wip

**Comment by jayvdb** [comment]
> 5:30:08 PM: CommandError: Failed to render view: 'total_count'

No doubt you are aware that this is the problem atm.

@sks444 , here is an example of the problem in distill.  It doesnt say which view caused the error.

File: unassigned_pr/unassigned_pr_scraper.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            commit_id = extracted_data[0]
+            url = 'https://api.github.com/search/issues?q='+commit_id + \
+                '+type:pr+repo:'+reponame  # Ignore PycodestyleBear (E501)
+            r = requests.get(url, headers={'Authentication':
+                                           'token '+GH_TOKEN,
+                                           'Accept':
+                                           'application/vnd.github.symmetra-preview+json'})  # Ignore PycodestyleBear (E501)  and LineLengthBear
+
+            # check commit is a opened as PR or not.
+            if r.json()['total_count'] > 0:
```
Comment (sks444) [inline unassigned_pr/unassigned_pr_scraper.py]: Can you show me the json you are getting with `r.json()`?

Comment (RaiVaibhav) [reply unassigned_pr/unassigned_pr_scraper.py]: The output is in this format
```
/home/vaibhav/test/backup.py
````

pastebin link https://pastebin.com/5DLEZBBQ

Comment (RaiVaibhav) [reply unassigned_pr/unassigned_pr_scraper.py]: Getting same error 
`CommandError: Failed to render view: 'total_count'`

Here is the netlify link of my fork. https://app.netlify.com/sites/raivaibhav-fork/deploys/5acc9ebfdd28ef5456638e1f

**Comment by RaiVaibhav** [comment]
:+1:  Thanks finally more fun :stuck_out_tongue:

File: unassigned_pr/unassigned_pr_scraper.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                a = r.json()['items'][0]['state']  # check PR is open or closed.
+                if a == 'open':
+                    # Ignore PycodestyleBear (E501)  and LineLengthBear and InvalidLinkBear
+                    url = 'https://api.github.com/repos/'+reponame+'/status/'+commit_id
```
Comment (Vamshi99) [inline unassigned_pr/unassigned_pr_scraper.py]: API url is wrong, I think you were looking for this https://developer.github.com/v3/repos/statuses/#get-the-combined-status-for-a-specific-ref

File: unassigned_pr/unassigned_pr_scraper.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import json
+import requests
+
+from github import Github
```
Comment (Vamshi99) [inline unassigned_pr/unassigned_pr_scraper.py]: Better to use IGitt instead of PyGithub

File: unassigned_pr/unassigned_pr_scraper.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+def unassigned_pr_json(request):
+    try:
+        GH_TOKEN = get_api_key('GH')
+    except Exception:
+        return HttpResponse('[]')
+    g1 = Github(GH_TOKEN)
+    org_name = get_org_name()
+    org = g1.get_organization(org_name)
+    repo = org.get_repo(org_name)
```
Comment (Vamshi99) [inline unassigned_pr/unassigned_pr_scraper.py]: It should be org.get_repo(repo_name)

File: unassigned_pr/unassigned_pr_scraper.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    issues = repo.get_issues()
+    issues_list = []
+    for myissue in issues:
+        if myissue.assignee is None:
+            label = []
+            for mylabel in myissue.labels:
+                label.append(mylabel.name)
+            if 'status/blocked' not in label:
```
Comment (Vamshi99) [inline unassigned_pr/unassigned_pr_scraper.py]: why to create new label list?
use `if 'status/blocked' not in myissue.labels:`

File: unassigned_pr/unassigned_pr_scraper.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        GH_TOKEN = get_api_key('GH')
+    except Exception:
+        return HttpResponse('[]')
+    g1 = Github(GH_TOKEN)
+    org_name = get_org_name()
+    org = g1.get_organization(org_name)
+    repo = org.get_repo(org_name)
+    issues = repo.get_issues()
+    issues_list = []
+    for myissue in issues:
```
Comment (Vamshi99) [inline unassigned_pr/unassigned_pr_scraper.py]: `myissue`->`issue`
`mylabel` -> `label`

**Comment by RaiVaibhav** [comment]
@Vamshi99  here is the IGitt Implementation https://github.com/coala/community/pull/128, please review this.

File: unassigned_pr/unassigned_pr_scraper.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        events = j.get_events()
+        data = []
+        for i in events:
+            if i.commit_id is not None:
+                data.append(str(i.commit_id))
+        extracted_data = data[-1:]
+        if len(extracted_data) == 1:
+            commit_id = extracted_data[0]
+            url = 'https://api.github.com/search/issues?q='+commit_id + \
+                '+type:pr+repo:'+reponame  # Ignore PycodestyleBear (E501)
```
[the remark this answers is not in the document]
Comment (nkprince007) [reply unassigned_pr/unassigned_pr_scraper.py]: you can use `str.format` method for ease.

**Comment by nkprince007** [comment]
IGitt docs are available at https://igitt.gitmate.io

**Comment by RaiVaibhav** [comment]
@nkprince007 https://github.com/coala/community/pull/128

Already opened the IGitt implemented PR

File: unassigned_pr/unassigned_pr_scraper.py (hunk 2 above)
Comment (RaiVaibhav) [reply unassigned_pr/unassigned_pr_scraper.py]: Vamsi It's working you can check here https://bpaste.net/show/b4bc3f120973

**Comment by RaiVaibhav** [comment]
@Vamshi99  @nkprince007  the main problem it script running locally but not with in `netlify` so after that as per the request of @sks444  I changed the implementation to IGitt in separate [ PR ](https://github.com/coala/community/pull/128)  and It running fine .

File: unassigned_pr/unassigned_pr_scraper.py (hunk 4 above)
Comment (jayvdb) [reply unassigned_pr/unassigned_pr_scraper.py]: yes, for clarity add `repo_name = org_name` with a comment explaining why

**Comment by RaiVaibhav** [comment]
@jayvdb I am failing to solve this issue related to `netlify`, currently IGitt will be the best one.

**Comment by RaiVaibhav** [comment]
Closing as implementation of same is already gets merged https://github.com/coala/community/pull/128

**Pull request closed without merging**